src/screens/BB-BUS/DetailScreen.js

```diff
@@ -57,326 +40,214 @@ const TEXT_BLOCKS_SORTER = {
+  });
+};
+
+const parseData = (data) => {
+  const snake_caseData = data?.directus?.service?.data;
+
+  if (!snake_caseData) return;
+
+  // workaround for having camelCase keys in `top10`
+  // GraphQL is returning snake_case, see: https://github.com/d12/graphql-remote_loader/issues/36
+  // transforming method thanks to: https://coderwall.com/p/iprsng/convert-snake-case-to-camelcase
+  return deepRenameKeys(snake_caseData, (key) => key.replace(/_\w/g, (m) => m[1].toUpperCase()));
+};
+
+const parseTextBlocks = (data) => {
+  const { textBlocks } = data[0];
+  let firstTextBlocks;
+  let sortedTextBlocks;
+
+  if (textBlocks) {
+    sortedTextBlocks = _sortBy(textBlocks, (textBlock) => {
+      // fix for multi nested result form Directus API
+      if (textBlock.textBlock) textBlock = textBlock.textBlock;
+
+      return TEXT_BLOCKS_SORTER[textBlock.name];
+    });
+
+    // filter text blocks we want to render before authorities and persons
+    firstTextBlocks = _remove(sortedTextBlocks, (textBlock) => {
+      // fix for multi nested result form Directus API
+      if (textBlock.textBlock) textBlock = textBlock.textBlock;
+
+      return (
+        textBlock.name.toUpperCase() === 'KURZTEXT' || textBlock.name.toUpperCase() === 'VOLLTEXT'
+      );
+    });
+
+    // filter text blocks, we do not want to render
+    _remove(sortedTextBlocks, (textBlock) => {
+      // fix for multi nested result form Directus API
+      if (textBlock.textBlock) textBlock = textBlock.textBlock;
+
+      return (
+        textBlock.name.toUpperCase() === 'FACHLICH FREIGEGEBEN DURCH' ||
+        textBlock.name.toUpperCase() === 'FACHLICH FREIGEGEBEN AM'
+      );
+    });
+  }
+
+  return { firstTextBlocks, sortedTextBlocks };
+};
+
+// eslint-disable-next-line complexity
+export const DetailScreen = ({ route }) => {
   const scrollViewRef = useRef();
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
-  const { orientation, dimensions } = useContext(OrientationContext);
-  const [refreshTime, setRefreshTime] = useState();
-  const [accordion, setAccordion] = useState({});
   const [showBackToTop, setShowBackToTop] = useState(false);
   const [refreshing, setRefreshing] = useState(false);
+
   const rootRouteName = route.params?.rootRouteName ?? '';
   const headerTitle = route.params?.title ?? '';
   const details = route?.params?.data ?? '';
   const id = details.id;
 
-  if (!id) return null;
-
   useMatomoTrackScreenView(matomoTrackingString([MATOMO_TRACKING.SCREEN_VIEW.BB_BUS, headerTitle]));
 
-  useEffect(() => {
-    const getRefreshTime = async () => {
-      const time = await refreshTimeFor(`BBBUS-service-${id}`, consts.REFRESH_INTERVALS.BB_BUS);
+  // TODO: why are there two different names for the web screens?
+  const openWebScreen = useOpenWebScreen(headerTitle, undefined, rootRouteName);
+  const openWebScreenDetailName = useOpenWebScreen(details.name, undefined, rootRouteName);
```

Comment:
![Simulator Screen Shot - iPhone 11 Pro Max - 2021-06-22 at 17 57 59](https://user-images.githubusercontent.com/1942953/122960271-8c1fc180-d383-11eb-88d3-5b6ab9e0e7ea.png)

i thought that there were different titles maybe for the button on the top and links in accordion. but in my tests there were the same even in the old version.
is `headerTitle` always `details.name`?

Reply:
were we pass the information, both is the same:

```
title: bbBusService.name,
routeName: 'BBBUSDetail',
params: {
  title: bbBusService.name,
  query: '',
  queryVariables: {},
  rootRouteName: 'BBBUS',
  shareContent: {
    message: shareMessage(bbBusService)
  },
  data: bbBusService
}
```

`title` is `.name` of the object that is passed to `data`, so `headerTitle` equals `details.name`, right?!

Reply:
i agree